### PR TITLE
Dm 5201 product editor main thumbnail

### PR DESCRIPTION
**Pull request by PhilipDeFraties**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-5201

## Description - what does this code do?
Adds functionality to product editor allowing for the uploading and editing of `:main_display_image` for products

Includes significant refactor of the `ProductController#update` action to leverage new `SaveProductService` class to better encapsulate updating logic, which is a pattern already in place with `practice` logic.

## Testing done - how did you test it/steps on how can another person can test it 
1. go through the `product` editor and verify that all 3 forms work as expected by updating fields, adding and deleting attachments (main display image for the `/description` page and image attachments on the `multimedia` page.
2. where applicable ensure that cropping images works as expected and additional changes to other fields also persist when included in form submission alongside cropping, adding, and deletion of images or attachments.
3. verify you can delete the `main_display_image` for a product.
4. go to `/innovations/flow3/edit/introduction` and verify you can remove, add, and crop the "Thumbnail" image and the changes persist as expected.

## Screenshots, Gifs, Videos from application (if applicable)
Screenshot 2024-10-04 at 12 52 10 PM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Unit tests written (if applicable)
- [ ] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating JIRA issue
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/controllers/products_controller.rb [hunk 1]
```diff
@@ -75,14 +63,17 @@ def product_params
       :description,
       :item_number,
       :vendor,
-      :vendor_link,
       :duns,
       :shipping_timeline_estimate,
-      :price,
```
Comment (camillevilla) [inline app/controllers/products_controller.rb:81]: we'll want `:vendor_link` and `:price` here

File: app/controllers/products_controller.rb [hunk 2]
```diff
@@ -13,52 +13,40 @@ def intrapreneur
...
+      notice = service.product_updated ? "Product was successfully updated." : nil
+      next_page = params[:next] ? "#{Product::PRODUCT_EDITOR_NEXT_PAGE[submitted_page.to_sym]}_" : nil
+      redirect_to send("product_#{next_page}path", @product), notice: notice
+    elsif service.errors.any?
+      flash[:error] = service.errors.join(', ')
+      redirect_to send("product_#{submitted_page}_path", @product) || admin_product_path(@product)
     else
-      redirect_to product_path(@product), notice: notice
+      next_page = params[:next] ? "#{Product::PRODUCT_EDITOR_NEXT_PAGE[submitted_page.to_sym]}_" : nil
+      redirect_to send("product_#{next_page}path", @product)
```
Comment (camillevilla) [inline app/controllers/products_controller.rb:44]:
```suggestion
      redirect_to send("product_#{next_page}_path", @product)
```

File: app/controllers/products_controller.rb (hunk 1 above)
Comment (PhilipDeFraties) [reply app/controllers/products_controller.rb:81]: ah yea I think I missed that in the rebase, thanks for catching

File: lib/tasks/products.rake [hunk 3]
```diff
@@ -147,6 +154,19 @@ namespace :products do
...
+
+      if PRACTICE_PARTNER_MAPPING[origin]
+        practice_partner = PracticePartner.find_or_initialize_by(name: PRACTICE_PARTNER_MAPPING[origin])
+        PracticePartnerPractice.create!(innovable: product, practice_partner: practice_partner)
+      end
+
+      vha_practice_partner = PracticePartner.find_or_initialize_by(slug: "vha-innovators-network")
+      PracticePartnerPractice.create!(innovable: product, practice_partner: vha_practice_partner)
+
+      puts "Created Product - #{product.name}"
```
Comment (camillevilla) [inline lib/tasks/products.rake:169]: I think a chunk of this got duplicated and ended up outside of the CSV block

File: app/models/practice.rb [hunk 4]
```diff
@@ -25,7 +25,6 @@ class Practice < Innovation
   attr_accessor :last_month_commits
   attr_accessor :two_months_ago_commits
   attr_accessor :three_months_ago_commits
-  attr_accessor :delete_main_display_image
   attr_accessor :crop_x, :crop_y, :crop_w, :crop_h
```
Comment (camillevilla) [inline app/models/practice.rb:28]:
```suggestion
  attr_accessor :crop_x, :crop_y, :crop_w, :crop_h
```

Comment (camillevilla) [reply app/models/practice.rb:28]: Looks like this can get moved up to `Innovation.rb` too!

**Review by camillevilla — APPROVED** [review]

**Pull request merged**